vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -422,6 +414,37 @@ public VimeoAccount authorizeWithClientCredentialsGrantSync() {
         return call;
     }
 
+    /**
+     * Synchronous version of {@link #singleSignOnTokenExchange(String, AuthCallback)}. This is useful
+     * when already running in a background thread, such as when using the Android AccountManager.
+     * See the documentation of {@link #singleSignOnTokenExchange(String, AuthCallback)} for more info.
+     *
+     * @param token           the authenticated user access token from application A. This <b>cannot</b> be a client
+     *                        credentials access token.
+     * @param accountName     the name of the account, usually the email
+     * @param basicAuthHeader a "basic" auth header to pass in; the basic auth header should be from
+     *                        the application that needs the new token. In the case of the Android
+     *                        account authenticator, which runs on the first app installed, we need
+     *                        to pass in that token
+     * @return A {@link VimeoAccount} if the sign on worked, or null
+     */
+    @Nullable
+    public VimeoAccount singleSignOnTokenExchange(@NotNull String token, @NotNull String accountName,
+                                                  @Nullable String basicAuthHeader) {
+
+        if (basicAuthHeader == null) {
+            basicAuthHeader = getBasicAuthHeader();
+        }
+        Call<VimeoAccount> call =
+                mVimeoService.ssoTokenExchange(basicAuthHeader, token, mConfiguration.scope);
+
+        VimeoAccount vimeoAccount = executeAccountCall(call);
+
+        saveAccount(vimeoAccount, accountName);
```

Comment:
It seems a little confusing that we have named the parameter email in some places and name in others. Does it have to be an email? If not, maybe we should rename all of the parameters as accountName instead.

Also, why does this method save the account too? Shouldn't that be left to the caller? If it remains here, maybe the null check of the client account should be at this level?